packages/flutter_tools/lib/src/runner/flutter_command_runner.dart

```diff
@@ -242,9 +245,59 @@ class FlutterCommandRunner extends CommandRunner<void> {
       }
     }
 
+    _machineFlagPresentInAnyCliArg = args.contains('--${FlutterGlobalOptions.kMachineFlag}');
     return super.run(args);
   }
 
+  /// Whether to perform a flutter version check, which prints a warning if old.
```

Comment:
Nit: Should a function's doc string explain (and thus be coupled to) what the caller(s) plan to do with it? (Not a big deal here since this comment is unlikely to become incorrect)

Reply:
Done